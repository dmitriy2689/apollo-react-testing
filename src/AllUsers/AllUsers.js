import React from 'react';
import { Link } from 'react-router-dom';
import { Query } from 'react-apollo';
import query from './AllUsers.query';

const AllUsers = () => (
  <Query query={query}>
    {({ loading, error, data }) => {
      if (loading) return <p>Loading...</p>
      if (error) return <p>Error!</p>

      return (
        <ul key='allUsers'>
          {data.allUsers.map(({ id, name }) => (
            <li key={id}>
              <Link to={`/user/${id}`}>
                {name ? name : 'incognoito'}>
              </Link>
            </li>
          ))}
        </ul>
      );
    }}
  </Query> 
)

export default AllUsers;