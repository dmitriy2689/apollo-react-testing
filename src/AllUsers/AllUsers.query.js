import gql from 'graphql-tag';

const query = gql`
  query {
    allUsers {
      id
      name
    }
  }
`;

export default query;