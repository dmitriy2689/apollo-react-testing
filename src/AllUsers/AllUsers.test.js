import React from 'react';
import { MockedProvider } from "react-apollo/test-utils";
import { BrowserRouter } from 'react-router-dom';
import { shallow } from 'enzyme';
import wait from "waait";
import AllUsers from './AllUsers';
import query from './AllUsers.query';

it('should render users list', async () => {
  const mock = {
    request: { query },
    result: { data: { allUsers: [
      { id: "1", name: "name1" },
      { id: "2", name: "name2" }
    ]}}
  };

  const wrapper = shallow(
    <BrowserRouter>
      <MockedProvider mocks={[mock]} addTypename={false}>
        <AllUsers />
      </MockedProvider>
    </BrowserRouter>
  );

  await wait(0);

  expect(wrapper).toMatchSnapshot();
})
