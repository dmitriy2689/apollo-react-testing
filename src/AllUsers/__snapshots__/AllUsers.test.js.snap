// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render users list 1`] = `
<Router
  history={
    Object {
      "action": "POP",
      "block": [Function],
      "createHref": [Function],
      "go": [Function],
      "goBack": [Function],
      "goForward": [Function],
      "length": 1,
      "listen": [Function],
      "location": Object {
        "hash": "",
        "pathname": "/",
        "search": "",
        "state": undefined,
      },
      "push": [Function],
      "replace": [Function],
    }
  }
>
  <MockedProvider
    addTypename={false}
    mocks={
      Array [
        Object {
          "request": Object {
            "query": Object {
              "definitions": Array [
                Object {
                  "directives": Array [],
                  "kind": "OperationDefinition",
                  "name": undefined,
                  "operation": "query",
                  "selectionSet": Object {
                    "kind": "SelectionSet",
                    "selections": Array [
                      Object {
                        "alias": undefined,
                        "arguments": Array [],
                        "directives": Array [],
                        "kind": "Field",
                        "name": Object {
                          "kind": "Name",
                          "value": "allUsers",
                        },
                        "selectionSet": Object {
                          "kind": "SelectionSet",
                          "selections": Array [
                            Object {
                              "alias": undefined,
                              "arguments": Array [],
                              "directives": Array [],
                              "kind": "Field",
                              "name": Object {
                                "kind": "Name",
                                "value": "id",
                              },
                              "selectionSet": undefined,
                            },
                            Object {
                              "alias": undefined,
                              "arguments": Array [],
                              "directives": Array [],
                              "kind": "Field",
                              "name": Object {
                                "kind": "Name",
                                "value": "name",
                              },
                              "selectionSet": undefined,
                            },
                          ],
                        },
                      },
                    ],
                  },
                  "variableDefinitions": Array [],
                },
              ],
              "kind": "Document",
              "loc": Object {
                "end": 56,
                "start": 0,
              },
            },
          },
          "result": Object {
            "data": Object {
              "allUsers": Array [
                Object {
                  "id": "1",
                  "name": "name1",
                },
                Object {
                  "id": "2",
                  "name": "name2",
                },
              ],
            },
          },
        },
      ]
    }
  >
    <AllUsers />
  </MockedProvider>
</Router>
`;
